extension/src/repository/index.ts

```diff
@@ -51,34 +52,23 @@ export class Repository extends DeferredDisposable {
     return this.model.getChildren(path)
   }
 
-  public update(path?: string) {
-    return this.data.managedUpdate(path)
+  public update() {
+    return this.data.managedUpdate()
   }
 
   public hasChanges(): boolean {
-    return this.model.hasChanges()
+    return this.model.getHasChanges()
   }
 
   public getScale() {
-    return { tracked: this.model.getDecorationState().tracked.size }
-  }
-
-  public setExperiments(experiments: Experiments) {
-    this.dispose.track(
-      experiments.onDidChangeExperiments(data => {
-        if (data) {
-          this.model.transformAndSetExperiments(data)
-          this.setState()
-        }
-      })
-    )
+    return { tracked: this.model.getScale() }
   }
 
   private async initialize() {
     this.dispose.track(
       this.data.onDidUpdate(data => {
-        this.model.setState(data)
-        this.setState()
+        const state = this.model.setState(data)
+        this.setState(state)
```

Comment:
[F] Doing this means we don't have to hold all of the decoration and source control management state in (at least) two places.